pkg/buildpack/multi_architecture_helper.go

```diff
@@ -0,0 +1,170 @@
+package buildpack
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/paths"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/pkg/dist"
+	"github.com/buildpacks/pack/pkg/logging"
+)
+
+type MultiArchConfig struct {
+	buildpackTargets []dist.Target
+	expectedTargets  []dist.Target
+	logger           logging.Logger
+}
+
+func NewMultiArchConfig(targets []dist.Target, expected []dist.Target, logger logging.Logger) (MultiArchConfig, error) {
+	// Let's do some validations
+	return MultiArchConfig{
+		buildpackTargets: targets,
+		expectedTargets:  expected,
+		logger:           logger,
+	}, nil
+}
+
+func (m *MultiArchConfig) Targets() []dist.Target {
+	if len(m.expectedTargets) == 0 {
+		return m.buildpackTargets
+	}
+	return m.expectedTargets
+}
+
+// CopyConfigFiles Given a base directory, which is expected to be the root folder of single buildpack,
+// it will copy a buildpack.toml file, for each target, into the corresponding platform root folder.
+func (m *MultiArchConfig) CopyConfigFiles(baseDir string) ([]string, error) {
+	var filesToClean []string
+	for _, target := range m.Targets() {
+		// TODO we are not handling distributions versions yet
+		path, err := CopyConfigFile(baseDir, target, "")
+		if err != nil {
+			return nil, err
+		}
+		if path != "" {
+			filesToClean = append(filesToClean, path)
+		}
+	}
+	return filesToClean, nil
+}
+
+// PrepareDependencyConfigFile when creating a composite buildpack, dependencies URI are relative to the main buildpack
+// if we are building a multi-arch composite buildpack we MAY need to copy the buildpack.toml file to the dependency and
+// determine the platform root folder. This method will do those operations and return the path to the buildpack.toml that
+// was copied.
+func PrepareDependencyConfigFile(baseDir, depURI string, target dist.Target, version string, failWhenURI bool) (string, error) {
+	// Only in cases it is a URILocator we may want to copy config files
+	locatorType, err := GetLocatorType(depURI, baseDir, []dist.ModuleInfo{})
+	if err != nil {
+		return "", err
+	}
+
+	if locatorType == URILocator {
+		if failWhenURI {
```

Comment:
AFAIK this is true when `len(targets) > 1 && (opts.Publish || opts.Format == FormatFile)`. What does that have to do with composite buildpacks?

Reply:
I asked this in a WG session, today you can have your Buildpacks defined on your filesystem and use `pack buildpack package` to create the OCI artifact. In the multi-platform scenario, if you try to do that it is a little tricky because how do we define different URI for each platform? so we decided not to allow this, you need to push your dependencies to a registry (registry will have the index to handle the complexity). that's the reason of that boolean

Reply:
I removed this code, it is not required anymore